### PR TITLE
Extract common library and reorganize namespaces

### PR DESCRIPTION
**Pull request by tshemsedinov**
Refs: https://github.com/metarhia/Metarhia/issues/2, https://github.com/metarhia/MetaSync/issues/36

### CODE REVIEW

File: gs.js [hunk 1]
```diff
@@ -2,26 +2,28 @@
...
 
-gs.Connection = require('./connection.js');
-gs.Category = require('./category.js');
-gs.transformations = require('./transformations.js');
+gs.Connection = require('./lib/connection.js');
+gs.Category = require('./lib/category.js');
+gs.transformations = require('./lib/transformations.js');
+
+const api = {};
+api.common = require('metarhia-common');
```
Comment (aqrln) [inline gs.js]: Unused import

File: lib/cursor.mongodb.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
 'use strict';
 
-const util = require('util');
+const api = {};
+api.util = require('util');
+
 const transformations = require('./transformations.js');
```
Comment (aqrln) [inline lib/cursor.mongodb.js]: Is it intentional that some required modules are put into `api` and some aren't?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "globalstorage",
-  "version": "0.1.3",
+  "version": "0.1.5",
```
Comment (aqrln) [inline package.json]: Why `0.1.5`? `0.1.3` seems to be the last published version.

File: test/examples.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
 'use strict';
 
+const api = {};
 const gs = require('..');
-const metasync = require('metasync');
-const mongodb = require('mongodb').MongoClient;
+api.metasync = require('metasync');
```
Comment (aqrln) [inline test/examples.js]: `metasync` usage at line 53 must be updated to use the namespace.

File: lib/cursor.mongodb.js (hunk 2 above)
Comment (tshemsedinov) [inline lib/cursor.mongodb.js]: Sure, all should be in namespaces

Comment (tshemsedinov) [inline lib/cursor.mongodb.js]: ...but in this certain place `transformations` is a part of current module, internal/submodule, no need to put it inside api. I'll put external requires into api.* for consistence with Impress namespaces.

Comment (aqrln) [inline lib/cursor.mongodb.js]: That's what I asked about.

File: README.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
 # GlobalStorage
 
 [![TravisCI](https://travis-ci.org/metarhia/GlobalStorage.svg?branch=master)](https://travis-ci.org/metarhia/GlobalStorage)
-[![bitHound](https://www.bithound.io/github/metarhia/GlobalStorage/badges/score.svg)](https://www.bithound.io/github/metarhia/GlobalStorages)
+[![bitHound](https://www.bithound.io/github/metarhia/GlobalStorage/badges/score.svg)](https://www.bithound.io/github/metarhia/GlobalStorage)
 [![NPM Version](https://badge.fury.io/js/globalstorage.svg)](https://badge.fury.io/js/globalstorage)
-[![NPM Downloads/Month](https://img.shields.io/npm/dm/globalstorage.svg?style=flat-square)](https://www.npmjs.com/package/globalstorage)
-[![NPM Downloads](https://img.shields.io/npm/dt/globalstorage.svg?style=flat-square)](https://www.npmjs.com/package/globalstorage)
+[![NPM Downloads/Month](https://img.shields.io/npm/dm/globalstorage.svg)](https://www.npmjs.com/package/globalstorage)
+[![NPM Downloads](https://img.shields.io/npm/dt/globalstorage.svg)](https://www.npmjs.com/package/globalstorage)
```
Comment (belochub) [inline README.md]: In my opinion, all the changes made to README.md file should be made in another commit or at least listed in full commit description.

Comment (tshemsedinov) [inline README.md]: Yes, we have such f792ccf60aa7dff9883c7d738704f9a1a77b8330 I added this changes in this PR by mistake

**Comment by tshemsedinov** [comment]
Landed in 03a21109e9226c3396cf9696ae422548418ef30b
